extension/src/plots/model/index.ts

```diff
@@ -154,9 +154,7 @@ export class PlotsModel extends ModelWithPersistence {
   }
 
   public getCustomPlots(): CustomPlotsData | undefined {
-    const experiments = this.experiments
-      .getWorkspaceCommitsAndExperiments()
-      .filter(({ id }) => id !== EXPERIMENT_WORKSPACE_ID)
+    const experiments = this.experiments.getUnfilteredCommitsAndExperiments()
```

Comment:
[I] Need to handle the case where there are no unfiltered experiments left. IIRC we will currently show an empty plot.

Reply:
If I understand your statement correctly, when I tested for this possibility I got the "No Plots To Display" message which I thought worked fine. 

<img width="1266" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/2bc03614-1c71-4283-b8de-599704cf9b70">

Though we could extend the description for more clarity 🤔. For example, "No Plots to Display. Are any of your experiments unfiltered?"

Reply:
Updated the description:

<img width="936" alt="image" src="https://github.com/iterative/vscode-dvc/assets/43496356/252d92c8-10b4-4fbc-9e37-9a43d65bfca3">

Reply:
There should be a different state depending on whether or not there are custom plots created (there will always be data unless all of the experiments are filtered).

Reply:
I.e no plots created is different from no unfiltered data

Reply:
Right now we have two states, "no data" (No Plots to Display. Are any...) and "no plots created"(No Plots Added). But you're saying we should introduce a third separate state for when there are no unfiltered experiments? Would be easy enough :)

Reply:
What I am trying to avoid is this being brittle/getting forgotten about when we make changes because we have had issues with stale empty states previously.

```
  if (!hasData) {
    return <EmptyState isFullScreen={false}>No Plots to Display</EmptyState>
  }

  if (order.length === 0) {
    return <EmptyState isFullScreen={false}>No Plots Added</EmptyState>
  }
```

Previously `!hasData` was a placeholder because we could never hit that state. This was because there always had to be at least one commit in the table which would produce a single datapoint for the custom plots. This is almost an error state. 

IMO we should be explicit to the user what is going on. I.e. if we know there are plots but no data to put in them we should state something like `There is no data to Plot`. 

You cannot separate out the messages in the current state, i.e. `!hasData` will happen if all of the experiments are filtered whether plots have been added or not. We need to perform the `order.length` check before sending the webview message and send that information instead of `undefined`.

You could replace the `No Plots to Display.` text with `There is No Data to Plot` but you would need to add tests on both sides of the message to document that behaviour and make sure that in future when we change the code we know what the expected flow is. Relying on the assumption of "if there is no data then all the experiments must be filtered" seems optimistic at best.

Reply:
OHHHH, ok I see what you're saying now 🤦‍♀️ . Thanks for the further explanation and apologies, I don't know why my brain was struggling so much here 😅. 

I'll update the backend to send a different kind of data when there is no unfiltered values so that the webview can detect the difference and updates tests accordingly.